contracts/src/v0.7/dev/KeeperRegistryDev.sol

```diff
@@ -50,9 +57,9 @@ contract KeeperRegistryDev is
   uint64 private constant UINT64_MAX = 2**64 - 1;
   uint96 private constant LINK_TOTAL_SUPPLY = 1e27;
 
-  uint256 private s_upkeepCount;
-  uint256[] private s_canceledUpkeepList;
+  uint256 private s_nonce;
   address[] private s_keeperList;
+  EnumerableSet.UintSet private s_upkeepIDs;
```

Comment:
The change here of cancelled list to uncanceled list is pretty obvious, but the type change comes with a subtle behavior change: [the list now gets reordered](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol#L90).

If you restart after an upkeep has been cancelled, then the order will have changed compared to someone that originally read it before the upkeep was cancelled.Given we're moving to hashes, they won't really appear ordered.

Will a list that changes in order affect the off-chain bucketing algo, or do we take that into account somehow already?

Reply:
@se3000 
reordering won't affect the off chain bucketing algo. I did make [a note to be weary of order changes here](https://github.com/smartcontractkit/chainlink/blob/c225579595a546b3c4194fca23382b83aeb771f3/contracts/src/v0.7/dev/KeeperRegistryDev.sol#L571).